src/main/java/org/checkerframework/specimin/UnsolvedSymbolVisitor.java

```diff
@@ -743,6 +744,48 @@ public Visitable visit(BlockStmt node, Void p) {
     return result;
   }
 
+  @Override
+  public Visitable visit(InstanceOfExpr node, Void p) {
+    // If we have x : X and x instanceof Y, then X must be a supertype
+    // of Y if X != Y. The JLS says (15.20.2): "If a cast of the RelationalExpression to the
+    // ReferenceType would be rejected as a compile-time error, then the instanceof relational
+    // expression likewise produces a compile-time error. In such a situation, the result of the
+    // instanceof expression could never be true."
+    //
+    // This visit method uses this fact to add extends clauses to classes created by
+    // UnsolvedSymbolVisitor.
+    if (node.getPattern().isPresent()) {
+      // TODO: handle JDK14 pattern variables
+      return super.visit(node, p);
+    }
+    ReferenceType referenceType = node.getType();
+    Expression relationalExpr = node.getExpression();
+    String relationalExprFQN, referenceTypeFQN;
+    try {
+      referenceTypeFQN = referenceType.resolve().describe();
+      relationalExprFQN = relationalExpr.calculateResolvedType().describe();
+    } catch (UnsolvedSymbolException e) {
+      // Try again next time.
+      this.gotException();
+      return super.visit(node, p);
+    }
+
+    if (referenceTypeFQN.equals(relationalExprFQN)) {
+      // A type can't extend itself.
+      return super.visit(node, p);
+    }
+
+    for (UnsolvedClassOrInterface syntheticClass : missingClass) {
+      if (syntheticClass.getQualifiedClassName().equals(referenceTypeFQN)) {
```

Comment:
Are we sure we cannot get a null pointer from syntheticClass.getQualifiedClassName()?
If so, it makes sense to do this:
```
    String qualifiedClassName = syntheticClass.getQualifiedClassName();
    if (qualifiedClassName != null && qualifiedClassName.equals(referenceTypeFQN)) {
```

Otherwise, this is fine.

Reply:
> Are we sure we cannot get a null pointer from syntheticClass.getQualifiedClassName()?

Yes, we're sure. Specimin uses the Checker Framework's [Nullness Checker](https://checkerframework.org/manual/#nullness-checker), so the type system would warn us if `getQualifiedName()` could return an `@Nullable String`.